java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -86,21 +87,25 @@ public Pop(ProgramState programState, List<SymbolicValue> result) {
   final PMap<Symbol, SymbolicValue> values;
   final PMap<SymbolicValue, Constraint> constraints;
 
+  @Nullable
+  private MethodYield selectedYield;
+
   private ProgramState(PMap<Symbol, SymbolicValue> values, PMap<SymbolicValue, Integer> references,
     PMap<SymbolicValue, Constraint> constraints, PMap<ExplodedGraph.ProgramPoint, Integer> visitedPoints,
-    Deque<SymbolicValue> stack, SymbolicValue exitSymbolicValue) {
+    Deque<SymbolicValue> stack, SymbolicValue exitSymbolicValue, @Nullable MethodYield selectedYield) {
     this.values = values;
     this.references = references;
     this.constraints = constraints;
     this.visitedPoints = visitedPoints;
     this.stack = stack;
     this.exitSymbolicValue = exitSymbolicValue;
+    this.selectedYield = selectedYield;
     constraintSize = 3;
   }
   private ProgramState(Symbol symbol, PMap<Symbol, SymbolicValue> values, PMap<SymbolicValue, Integer> references,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)